### DISCUSSION TITLE
Clarification on the role of some external (required) components (postgres,rabbitmq,redis)

### DISCUSSION THREAD

**Discussion by Willena** [ONLYOFFICE/DocumentServer#1739] [Q&A] [answered]
Hi, 

OnlyOffice requires other software component to work:
- PostgreSql
- Redis
- RabbitMq

I couldn't find in the documentation the answer to the following question (for `RabbitMq`, `Redis`, `PostgreSql`)
- What the component is used for
- What kind of data is stored (if any)
- Who is producing / consuming the data
- How important the component is (heavy usage, small usage, critical data, ...)

Thanks for your time.

**Comment by ShockwaveNN** [comment, marked as answer]
Hi, we do not have any official documentation but If I remember right

* Postgreqsl and rabbitmq is used to store temp data needed for CoEdit service for work and build files after all user are exit
* redis is used to store cache file. And it's not used in open source version of DocumentServer

> Who is producing / consuming the data

I think it's better to search through source code and find calls for those subsystems

> How important the component is (heavy usage, small usage, critical data, ...)

It depends on scenarios your DocumentServer is used. For example if you got 1000 separate document in which only 1 user is opening it - there will be small usage of those subsystems

But if you got a single document with 1000 user in same document - I think usage is rather heavy
